docs/developer/validation-rules/resource-model-validation.md

```diff
@@ -0,0 +1,39 @@
+# Identifying resource model definitions in a specification
+## Description
+A model definition is considered as a resource model definition if:
+
+- The model is returned by a PUT operation as a part of 200 or 201 response codes.
+- The model has an 'allOf' Resource at any level [name of the root model is 'Resource'] AND the model is returned by a GET operation in 200 response.
+- The model has 'x-ms-azure-resource': true or if any model has an 'x-ms-azure-resource' set to true in its model hierarchy and is not named either of 'Resource', 'TrackedResource' or 'ProxyResource'.
+
+## Types of resource models
+Any resource can be categorized into 2 types: `TrackedResources` and `ProxyResources`.
+1. TrackedResource: A resource is considered a tracked resource if a model hierarchy has a property named 'location' which is marked as __required__.
+2. ProxyResource: Proxy resources are all resources without a required 'location' property in their model hierarchy.
+
+## Validation rules for resource models
+
+### M3007 - PutGetPatchResponseInvalid 
```

Comment:
@vishrutshah I'm not sure if we have assigned the right Id for this validation rule (`PutGetPatchResponseInvalid`), can you confirm this since you implemented it.